traits.yml

```diff
@@ -84,10 +145,210 @@ traits:
+          evaluated during processing of a three-dimensional scene.
+        usage:
+          - entity
+      IESProfile:
+        description: >
+          The entity represent an IES lighting profile, as defined by
+          IES-LM-63, see:
+
+              https://store.ies.org/product/lm-63-19-approved-method-ies-standard-file-format-for-the-electronic-transfer-of-photometric-data-and-related-information
+      SourcePath:
+        description: >
+          Defines a singular path within the entity's scene
+          hierarchy that should be used as the root, ignoring any other
+          peer paths.
+
+
+          Paths are delimited by a forward slash (/) and are always
+          considered absolute from the root of the entities scene
+          hierarchy.
+
+
+          For example "/cameras/1103/camera"
+      SourcePaths:
+        description: >
+          Defines the root paths within the entity data that form ths
+          start of a discreet hierarchy, labelled by the property key.
+
+
+          For example:
+            left: "/cameras/1103/camera/left"
+            right: "/cameras/1103/camera/right"
+  timeDomain:
+    description: >
+      Traits that are concerned with describing how data changes or is
+      parameterised over time.
+    members:
+      FrameRanged:
+        description: The data varies over a number of discrete frames.
+        usage:
+          - entity
+        properties:
+          startFrame:
+            description: >
+              The first frame covered by the entity data (inclusive).
+            type: integer
+          endFrame:
+            description: >
+              The last frame covered by the entity data (inclusive).
+            type: integer
+          inFrame:
+            description: >
+              The first frame to be used within startFrame/endFrame
+              (inclusive).
+            type: integer
+          outFrame:
+            description: >
+              The last frame to be used within startFrame/endFrame
+              (inclusive).
+            type: integer
+          framesPerSecond:
+            description: >
+              The number of frames that occur within one second.
+            type: float
+          step:
+            description: >
+              The interval between frames of the entity's data. A
+              startFrame of 21 and a step of 10 would indicate that the
+              entity data only covers frames 21, 31, 41, ...
+            type: integer
+      Static:
+        description: The data does not vary over time.
+        usage:
+          - entity
+  twoDimensional:
+    description: >
+      Traits that are concerned with describing two-dimensional data,
+      more colloquially known as 'images'.
+    members:
+      Image:
+        description: >
+          A family trait that should be composed with more specific
+          traits for any entity that holds two-dimensional data.
```

Comment:
Do you have any thoughts on what specifically the schema would capture?

Reply:
I was thinking it could be another usage trait, like `EntityFamily`. It's something I would be pro if we were making family traits a required concept. I don't think they are at the moment, but the wording of the docs could be read that they are. (It's another "should" thing...)